backend/fms_core/viewsets/sample_pooled.py

```diff
@@ -0,0 +1,117 @@
+
+class BiosampleSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Biosample
+        exclude = EXCLUDE_BOOKKEEPING
+
+class LibrarySerializer(serializers.ModelSerializer):
+    library_type = serializers.CharField(read_only=True, source='library_type.name')
+    platform = serializers.CharField(read_only=True, source='platform.name')
+    index = IndexSerializer(read_only=True, required=False)
+    class Meta:
+        model = Library
+        exclude = EXCLUDE_BOOKKEEPING
+        depth = 1
+
+class ProjectSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Project
+        fields = ['name', 'id']
+
+class DerivedSampleSerializer(serializers.ModelSerializer):
+    library = LibrarySerializer(required=False)
+    biosample = BiosampleSerializer(required=False)
+    tissue_source = SampleKindSerializer(required=False)
+    sample_kind=SampleKindSerializer(required=False)
+    project = ProjectSerializer(required=False)
+    
+    class Meta:
+        model = DerivedSample
+        exclude = EXCLUDE_BOOKKEEPING
+        depth = 1
+        
+
+class PooledSampleSerializer(serializers.Serializer):
+    ''' Serializes a DerivedBySample object, representing a pooled sample. 
+        The result is a nested data structure containing all of the information about the
+        pooled sample needed by the frontend for display.
+    '''
+    volume_ratio = serializers.DecimalField(max_digits=20, decimal_places=3, read_only=True)
+    derived_sample = DerivedSampleSerializer()
+    # Since DerivedBySample doesn't have its own id field, we use the derived_sample id
+    # as a top level id in the returned data structure. The UX needs this for 'objectsById' stuff.
+    id = serializers.IntegerField(read_only = True, source='derived_sample.id')
+    parent_sample_name=serializers.SerializerMethodField()
+    parent_sample_id = serializers.SerializerMethodField()
+    class Meta:
+        model = DerivedBySample
+        exclude = EXCLUDE_BOOKKEEPING + ['sample']
+
+    # Finds the id of the parent sample from which this pooled sample was derived. For example, if this
+    # pool member is from a library then it returns the id of the library sample. 
+    def get_parent_sample_id(self, obj):
+        parent_sample = Sample.objects.get(parent_of=obj.sample.id, derived_samples=obj.derived_sample.id)
+        return parent_sample.id if parent_sample is not None else ''
+
+    # Finds the id of the parent sample from which this pooled sample was derived.
+    def get_parent_sample_name(self, obj):
+        sample = Sample.objects.get(parent_of=obj.sample.id, derived_samples=obj.derived_sample.id)
+        return sample.name if sample is not None else ''
+
+
+class MissingPoolIDException(APIException):
+    status_code = 400
+    default_code = 'bad_request'
+    default_detail = 'No pool ID: query must include sample__id parameter containing the pool ID.'
+
+class PooledSamplesViewSet(viewsets.ModelViewSet):
```

Comment:
Build a generic viewset that can be used for other purpose than the pool detail. We may have a request down the line to list pools and export them as a csv. The current viewset would not work... The get_queryset need to be usable by list and retrieve both. You should add this filtering to an overloaded retrieve instead. Also this bit of code is likely not useful since you could just add the sample_id or sample__id to the filterset_fields and have the filtering take care of this.